packages/app/src/provider/BackgroundLockProvider.tsx

```diff
@@ -1,29 +1,40 @@
-import type { PropsWithChildren } from 'react'
-
+import { TypedArrayEncoder } from '@credo-ts/core'
 import { useSecureUnlock } from '@package/secure-store/secure-wallet-key/SecureUnlockProvider'
+import { usePathname } from 'expo-router'
 import { useRouter } from 'expo-router'
-import { useEffect, useRef } from 'react'
+import { type PropsWithChildren, useEffect, useRef } from 'react'
 import { AppState, type AppStateStatus } from 'react-native'
 
-const BACKGROUND_TIME_THRESHOLD = 60000 // 60 seconds
+const BACKGROUND_TIME_THRESHOLD = 60000 // 1 minute
 
 export function BackgroundLockProvider({ children }: PropsWithChildren) {
   const router = useRouter()
+  const pathname = usePathname()
   const secureUnlock = useSecureUnlock()
   const backgroundTimeRef = useRef<Date | null>(null)
+  const lastPathRef = useRef<string | null>(null)
 
   useEffect(() => {
     const handleAppStateChange = (nextAppState: AppStateStatus) => {
       if (nextAppState === 'background' || nextAppState === 'inactive') {
         backgroundTimeRef.current = new Date()
+        lastPathRef.current = pathname
       } else if (nextAppState === 'active') {
         if (backgroundTimeRef.current) {
           const timeInBackground = new Date().getTime() - backgroundTimeRef.current.getTime()
 
           if (timeInBackground > BACKGROUND_TIME_THRESHOLD && secureUnlock.state === 'unlocked') {
             console.log('App was in background for more than 30 seconds, locking')
+            const lastPath = lastPathRef.current
             secureUnlock.lock()
-            router.replace('/authenticate')
+
+            if (lastPath && !['/', '/authenticate'].includes(lastPath)) {
+              // Encode the path for redirection after unlock
+              const encodedRedirect = TypedArrayEncoder.toBase64URL(TypedArrayEncoder.fromString(lastPath))
+              router.replace(`/authenticate?redirectAfterUnlock=${encodedRedirect}`)
```

Comment:
I think the thing we want to fix is to handle a deeplinkg, but the wallet is closing the wallet on becoming active, so the deeplink isn't handled.

This may (not sure) fix the issue of a deeplink, but in case you were in a flow and went to the background, this won't work and require us to refactor a lot of of the logic, and will lead to bugs 🤔 

So i think we need to know here whether we are handling a deeplink or not. if not, we should just go to authenticate, if we do, we need to extract the deeplink url and redirect to that aftewards

Reply:
For the flow bug with the background lock; you could handle inactive differently than background. E.g. when the modal opens to do the authorization code flow we have a state of inactive, and we could allow for a longer time (e.g. 2-3 minutes) then when you manually close the app into the background (1 minute).